### PR TITLE
Handle SIGTERM for clean shutdown

### PR DESCRIPTION
**Pull request by sheilatron**
This is probably not yet ready to merge, but it's closer to a solution. It's possible we need to also add SIGTERM handling within the gevent server (not just the child process)

Tests pass when run without coverage, and we're seeing a clear log message that the SIGTERM is being handled:

2016-06-04 19:37:44 sphinxserve INFO Received SIGTERM signal to shut down!

...but there are still some errors when running with the following command, and the processes don't end cleanly.

coverage run --source=sphinxserve --concurrency multiprocessing --concurrency gevent -m py.test -c tests/pytest.ini tests --pdb

I will take a look at this again later in the week.

### CODE REVIEW

File: sphinxserve/__init__.py [hunk 1]
```diff
@@ -172,13 +173,17 @@ def manage(self):
...
         '''
         result = self.build()
-
         if result.code != 0:
             sys.exit(result.stderr)
-
         logger.debug(result.stdout)
         workers = [spawn(self.serve), spawn(self.watch), spawn(self.render)]
+
+        def shutdown_handler():
```
Comment (mzdaniel) [inline sphinxserve/__init__.py:181]: This code looks great.
Just a minor detail, moving this shutdown_handler function right after
the manage method docstring would read better.

**Comment by mzdaniel** [comment]
This looks great Sheila!
What errors are you getting?  What platform are you testing?

**Comment by mzdaniel** [comment]
Hey Sheila. Your PR is integrated in master now and testing coverage is working very well. Until Josh Williams's patch is in upstream, I created https://github.com/mzdaniel/coveragepy so we can properly test.

Thank you for your help at the Pycon sprint.

**Comment by sheilatron** [comment]
I'm glad to hear that test coverage reports are more accurate, now! Hopefully the fix to coverage will be merged upstream. Thanks @mzdaniel for mentoring me on this. During the sprint I learned some interesting things about signals, gevent, and multiprocessing.

**Comment by mzdaniel** [comment]
For the record, the coverage package now includes the work done at Pycon. This is how the new sphinxserve tests and coverage look like: 
(on linux py27 & py35) https://travis-ci.org/mzdanieltest/sphinxserve/jobs/158049239
(on OSX py27 & py35) https://travis-ci.org/mzdanieltest/sphinxserve/jobs/158049240

**Pull request closed without merging**